docs/content/configuration/configuration-reference.md

```diff
@@ -0,0 +1,1512 @@
+|`druid.indexer.runner.workerBlackListCleanupPeriod`|A duration after which the cleanup thread will startup to clean blacklisted workers.|PT5M|
+|`druid.indexer.runner.maxPercentageBlacklistWorkers`|The maximum percentage of workers to blacklist, this must be between 0 and 100.|20|
+
+There are additional configs for autoscaling (if it is enabled):
+
+|Property|Description|Default|
+|--------|-----------|-------|
+|`druid.indexer.autoscale.strategy`|Choices are "noop" or "ec2". Sets the strategy to run when autoscaling is required.|noop|
+|`druid.indexer.autoscale.doAutoscale`|If set to "true" autoscaling will be enabled.|false|
+|`druid.indexer.autoscale.provisionPeriod`|How often to check whether or not new middle managers should be added.|PT1M|
+|`druid.indexer.autoscale.terminatePeriod`|How often to check when middle managers should be removed.|PT5M|
+|`druid.indexer.autoscale.originTime`|The starting reference timestamp that the terminate period increments upon.|2012-01-01T00:55:00.000Z|
+|`druid.indexer.autoscale.workerIdleTimeout`|How long can a worker be idle (not a run task) before it can be considered for termination.|PT90M|
+|`druid.indexer.autoscale.maxScalingDuration`|How long the overlord will wait around for a middle manager to show up before giving up.|PT15M|
+|`druid.indexer.autoscale.numEventsToTrack`|The number of autoscaling related events (node creation and termination) to track.|10|
+|`druid.indexer.autoscale.pendingTaskTimeout`|How long a task can be in "pending" state before the overlord tries to scale up.|PT30S|
+|`druid.indexer.autoscale.workerVersion`|If set, will only create nodes of set version during autoscaling. Overrides dynamic configuration. |null|
+|`druid.indexer.autoscale.workerPort`|The port that middle managers will run on.|8080|
+
+### Overlord Dynamic Configuration
+
+The overlord can dynamically change worker behavior.
+
+The JSON object can be submitted to the overlord via a POST request at:
+
+```
+http://<OVERLORD_IP>:<port>/druid/indexer/v1/worker
+```
+
+Optional Header Parameters for auditing the config change can also be specified.
+
+|Header Param Name| Description | Default |
+|----------|-------------|---------|
+|`X-Druid-Author`| author making the config change|""|
+|`X-Druid-Comment`| comment describing the change being done|""|
+
+A sample worker config spec is shown below:
+
+```json
+{
+  "selectStrategy": {
+    "type": "fillCapacity",
+    "affinityConfig": {
+      "affinity": {
+        "datasource1": ["host1:port", "host2:port"],
+        "datasource2": ["host3:port"]
+      }
+    }
+  },
+  "autoScaler": {
+    "type": "ec2",
+    "minNumWorkers": 2,
+    "maxNumWorkers": 12,
+    "envConfig": {
+      "availabilityZone": "us-east-1a",
+      "nodeData": {
+        "amiId": "${AMI}",
+        "instanceType": "c3.8xlarge",
+        "minInstances": 1,
+        "maxInstances": 1,
+        "securityGroupIds": ["${IDs}"],
+        "keyName": ${KEY_NAME}
```

Comment:
Can you put ${KEY_NAME} in double-quotes with a comma after so it's proper JSON and doesn't get highlighted?

Reply:
Added quotes